core/src/main/java/org/apache/calcite/rel/type/RelDataTypeFactoryImpl.java

```diff
@@ -612,8 +612,15 @@ public SqlCollation getCollation() {
     }
 
     public SqlTypeName getSqlTypeName() {
+      Class clz =  clazz;
+      if (List.class.isAssignableFrom(clazz) || java.sql.Array.class.isAssignableFrom(clazz)
+          || java.lang.reflect.Array.class.isAssignableFrom(clazz)) {
+        clz = List.class;
+      } else if (Map.class.isAssignableFrom(clazz)) {
+        clz = Map.class;
+      }
       final SqlTypeName typeName =
-          JavaToSqlTypeConversionRules.instance().lookup(clazz);
+          JavaToSqlTypeConversionRules.instance().lookup(clz);
```

Comment:
Only `Map` is enough, in `JavaToSqlTypeConversionRules` we already have List and Array

Reply:
yes.